pkg/sitewise/datasource.go

```diff
@@ -112,55 +112,55 @@ func (ds *Datasource) HealthCheck(ctx context.Context, req *backend.CheckHealthR
 }
 
 func (ds *Datasource) HandleInterpolatedPropertyValueQuery(ctx context.Context, req *backend.QueryDataRequest, query *models.AssetPropertyValueQuery) (data.Frames, error) {
-	return ds.invoke(ctx, req, query.BaseQuery, func(ctx context.Context, sw client.SitewiseClient) (framer.Framer, error) {
+	return ds.invoke(ctx, req, &query.BaseQuery, func(ctx context.Context, sw client.SitewiseClient) (framer.Framer, error) {
 		return api.GetInterpolatedAssetPropertyValues(ctx, sw, *query)
 	})
 }
 
 func (ds *Datasource) HandleGetAssetPropertyValueHistoryQuery(ctx context.Context, req *backend.QueryDataRequest, query *models.AssetPropertyValueQuery) (data.Frames, error) {
-	return ds.invoke(ctx, req, query.BaseQuery, func(ctx context.Context, sw client.SitewiseClient) (framer.Framer, error) {
-		return api.BatchGetAssetPropertyValues(ctx, sw, *query)
+	return ds.invoke(ctx, req, &query.BaseQuery, func(ctx context.Context, sw client.SitewiseClient) (framer.Framer, error) {
+		return api.BatchGetAssetPropertyValues(ctx, sw, query)
```

Comment:
changed `*query` to `query`. this is to be able to add the AssetIds to the `query.BaseQuery` which is used later on when generating the frames:

https://github.com/grafana/iot-sitewise-datasource/blob/bee0f33038f7d829ff70f215ddcec4923f6dc7df/pkg/framer/property_value.go#LL18C38-L18C38

https://github.com/grafana/iot-sitewise-datasource/blob/bee0f33038f7d829ff70f215ddcec4923f6dc7df/pkg/resource/query_resouce_provider.go#L59

i wasn't sure if it's ok to do it this way by making the query mutatable.

Reply:
Hmm it seems like we shouldn't add any information to the query in those functions, but I cannot tell what could be a better solution.
Maybe adding something to the framer.Framer? (for example framer.AssetPropertyValue ?)

Reply:
I have a new idea to do this --
it's also going to change the query within the Handle... functions but we will manipulate and return just copies of the queries. 
We can also abandon using the "invoke" methods in those specific handlers. This way we don't need to modify the signature of "invoke."

Going to write up the suggestion.